src/storage/mvcc/txn.rs

```diff
@@ -47,21 +48,35 @@ impl<S: Snapshot> fmt::Debug for MvccTxn<S> {
 }
 
 impl<S: Snapshot> MvccTxn<S> {
-    pub fn new(
-        snapshot: S,
-        start_ts: u64,
-        mode: Option<ScanMode>,
-        isolation_level: IsolationLevel,
-        fill_cache: bool,
-    ) -> Self {
-        Self {
+    pub fn new(snapshot: S, start_ts: u64, fill_cache: bool) -> Result<Self> {
+        Ok(Self {
             // Todo: use session variable to indicate fill cache or not
-            reader: MvccReader::new(snapshot, mode, fill_cache, None, None, isolation_level),
+            // ScanMode is `None`, since in prewrite and other operations, keys are not given in
+            // order and we use prefix seek for each key. An exception is GC, which uses forward
+            // scan only.
+            // IsolationLevel is `SI`, actually the method we use in MvccTxn does not rely on
+            // isolation level, so it can be any value.
+            reader: MvccReader::new(
+                snapshot.clone(),
+                None,
+                fill_cache,
+                None,
+                None,
+                IsolationLevel::SI,
+            ),
+            gc_reader: MvccReader::new(
```

Comment:
So you will create two readers in every case, but only use one of them?

Reply:
Before Mvcc refinement PR is landed, yes. The cost should be negligible since we are not creating new cursors in `MvccReader::new()`.

Reply:
How about wrap it in an `Option` or something?

Reply:
you can use Either

Reply:
but if we will refine this soon, I am fine with this too, but you need to add some comments here.

Reply:
@MyonKeminta Do you mean that make it not optional when gc is called?